File: solutions/javascript/line-up/1/line-up.js
//
// This is only a SKELETON file for the 'Line Up' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const format = (name, position) => {
  const position_str = String(position);

  const position_str_end = position_str.slice(position_str.length - 2);
  if(position_str_end == '11' || position_str_end == '12' || position_str_end == '13'){
    return formatString(name, `${position}th`);
  }

  switch(position_str[position_str.length-1]){
    case '1':
      return formatString(name, `${position}st`);
    case '2':
      return formatString(name, `${position}nd`);
    case '3':
      return formatString(name, `${position}rd`);
    default:
      return formatString(name, `${position}th`);
  }
};
const formatString = (name, position) => {
  return `${name}, you are the ${position} customer we serve today. Thank you!`
}

File: solutions/csharp/line-up/1/LineUp.cs
public static class LineUp
{
    public static string Format(string name, int number)
    {

        string number_str = Convert.ToString(number);
        if(number_str.Length != 1)
        {
            string number_str_end2 = number_str.Substring(number_str.Length-2, 2);
            if (number_str_end2 == "11" || number_str_end2 == "12" || number_str_end2 == "13")
            {
                return FormatOutput(name, $"{number_str}th");
            }
        }
        
        

        string number_str_end1 = number_str.Substring(number_str.Length-1, 1);

        if(number_str_end1 == "1")
        {
            return FormatOutput(name, $"{number_str}st");
        }
        if(number_str_end1 == "2")
        {
            return FormatOutput(name, $"{number_str}nd");
        }
        if(number_str_end1 == "3")
        {
            return FormatOutput(name, $"{number_str}rd");
        }
        return FormatOutput(name, $"{number_str}th");
    }
    public static string FormatOutput(string name, string number_with_order)
    {
        return $"{name}, you are the {number_with_order} customer we serve today. Thank you!";    }
}
